### DISCUSSION TITLE
does the docking feature work under wayland

### DISCUSSION THREAD

**Discussion by shi-yan** [ocornut/imgui#5279] [New Users Build/Link/Run issues ONLY!]
From what I read from a few places, wayland doesn't allow querying for a window's absolute location with respect to the desktop.

If that's true, is the docking feature still work for wayland?

I want to shop for a GUI framework that can implement something similar to chrome's detachable tabs. I wonder if there is any hard limit on wayland.

Thanks,

**Comment by ocornut** [comment]
Docking works.
Multi-viewports, which is what you are asking for, doesn’t.

**Reply by shi-yan** [reply]
Thank you very much. Is this limited by wayland? because if this is, I will give up on supporting wayland.

Thanks.

**Reply by rokups** [reply]
Viewports work on X11. Reason why they do not work on wayland is that wayland does not provide absolute window coordinates and does not allow to explicitly position a window. Wayland may be supported some time in the (distant) future maybe. No promises, but we are definitely evaluating what could be done about it.

**Comment by ocornut** [comment]
Linux works but we have problems on some window managers. Windows and OSX tend to work well. You can find those answers if you browse the wiki.

You can still simply disable multi-viewports on Wayland.
